server/routers/viewer.tsx

```diff
@@ -28,7 +28,7 @@ export const viewerRouter = createProtectedRouter()
       const { prisma, user } = ctx;
       const bookingListingByStatus = input.status;
       const bookingListingFilters: Record<typeof bookingListingByStatus, Prisma.BookingWhereInput[]> = {
-        upcoming: [{ endTime: { gte: new Date() } }],
+        upcoming: [{ endTime: { gte: new Date() }, NOT: { status: { equals: BookingStatus.CANCELLED } } }],
         past: [{ endTime: { lte: new Date() } }],
         cancelled: [{ status: { equals: BookingStatus.CANCELLED } }],
       };
```

Comment:
Can't we just use the `input.status` in the `where` of `bookingsQuery` below?

I.e.

```ts
      const bookingsQuery = await prisma.booking.findMany({
        where: {
          ...(input.status === 'upcoming' ? { /* specific filters for upcoming */ } : {}),
          ...(input.status === 'past' ? { /* specific filters for past */ } : {}),
          ...(input.status === 'cancelled' ? { /* specific filters for cancelled */ } : {}),
      },
    // ...
```